pkg/plugin/test-keeper/plugin/event_handler.go

```diff
@@ -141,33 +131,44 @@ func (gh *GitHubTestEventsHandler) handlePrComment(prComment *gogh.IssueCommentE
 		RepoName: *prComment.Repo.Name,
 		Hash:     *pullRequest.Head.SHA,
 	}
-	statusService := gh.newTestStatusService(change)
 
 	if comment == SkipComment && *prComment.Action == "deleted" {
-		testsExist, err := gh.checkTests(change, *prNumber)
-		if err != nil {
-			return err
-		}
-		if testsExist {
-			return statusService.testsExist()
-		}
-
-		return statusService.noTests()
+		return gh.checkTestsAndSetStatus(change, pullRequest)
 	}
 
+	statusService := gh.newTestStatusService(change)
 	return statusService.okWithoutTests(*sender)
 }
 
-func (gh *GitHubTestEventsHandler) checkTests(change scm.RepositoryChange, prNumber int) (bool, error) {
-	configLoader := config.PluginConfigLoader{PluginName: ProwPluginName, Change: change}
+func (gh *GitHubTestEventsHandler) checkTestsAndSetStatus(change scm.RepositoryChange, pr *gogh.PullRequest) error {
+	url, configuration := LoadTestKeeperConfig(gh.Log, change)
+	testsExist, err := gh.checkTests(change, configuration, *pr.Number)
+	if err != nil {
+		return err
+	}
 
-	configuration := TestKeeperConfiguration{Combine: true}
-	err := configLoader.Load(&configuration)
+	statusService := gh.newTestStatusService(change)
+	if testsExist {
+		return statusService.testsExist()
+	}
+
+	err = statusService.noTests()
 	if err != nil {
-		gh.Log.Warnf("Config file was not loaded. Cause: %", err)
+		gh.Log.Error("There occur an error when the status was being set to PR:", err)
+	}
+
+	commentContext := plugin.CommentContext{PluginName: ProwPluginName, Assignee: *pr.User.Login}
+	commentService := plugin.NewCommentService(gh.Client, gh.Log, change, *pr.Number, commentContext)
+
+	cerr := commentService.PluginComment(CreateCommentMessage(url, configuration, change))
+	if cerr != nil {
+		return cerr
 	}
+	return err
+}
 
-	matcher := LoadMatcher(configuration)
+func (gh *GitHubTestEventsHandler) checkTests(change scm.RepositoryChange, config TestKeeperConfiguration, prNumber int) (bool, error) {
```

Comment:
Can't we have `config` as part of the `GitHubTestEventsHandler` struct?

Reply:
I was thinking about it as well - I wasn't sure if it wouldn't be confusing that when the `GitHubTestEventsHandler` struct is  instantiated, that it would require the config that is not available at that time